src/DynamoCore/UpdateManager/UpdateManager.cs

```diff
@@ -1023,8 +1023,62 @@ void client_DownloadProgressChanged(object sender, DownloadProgressChangedEventA
         /// </summary>
         internal static void CheckForProductUpdate()
         {
+            //If we already have higher version installed, don't look for product update.
+            if(new DynamoLookUp().LatestProduct > Instance.ProductVersion)
```

Comment:
In order to mock the registry specific calls, I have made those methods as virtual and hence I need an instance method/property to call those virtual methods that are mocked in test environment.